### DISCUSSION TITLE
[Show and Tell] MaterialFX rewrite, state as of 26/01/2023

### DISCUSSION THREAD

**Discussion by palexdev** [palexdev/MaterialFX#287] [Show and tell]
It's been a while since I last wrote an update like this to tell people how things are going, what am I doing, and so on.
It's also been a while since the last major update, I don't want to disappoint you but to be honest, the new version will still take a long time before even going into "beta" stage.
I don't know if anyone noticed, well it's also written in the title, I'm working on a re-write of MaterialFX with the following goals:
- Modularize the project. As of now, I've divided it in 5 modules: 'core', 'effects', 'localization', 'components', 'resources'. All of them will be hosted here in the same repository, will be easier to maintain and contribute to, each will have a different artifact name and will be published on Maven as separate artifacts
- Polish and cleanup the existing code where possible
- Introduce the Theme API, transitioning to Material Design 3 (will require a lot of time to implement properly)
- As a consequence of the above goal, stop using the `getUserAgentStylesheet()` method which troubles this project since the beginning
- Introduce a Behavior API that will let users customize the way components work (on the interaction 'side')
- ...and others that honestly I fail to remember now, there's a TON of work to do

Today, huge progress on the enhancement of the Icons API that is now... on steroids, check this:
![New_MFXFontIcon_demo](https://user-images.githubusercontent.com/16880178/214715354-b3ded2f7-8061-4eca-8fff-3b36186b9e7a.png)
I won't describe the API here and now, as always everything will be documented on release. You just have to know that now MFXFontIcon is capable of using different icon packs. The 'officially' supported ones will have a super simple API to switch between them. For the third party ones, the dev/user will have the responsibility to tell the icon how to convert an icon 'description' to the actual Unicode char (nothing complex, I assure you it's super easy).
Also, as you can see from the image, I decided to drop MFXResources icon pack in favor of FontAwesome (latest v6).

**Comment by HansITChange** [comment]
I'm looking forward to it !

**Comment by palexdev** [comment]
## Update
The new MFXResources module is now available in its first version on Maven
```xml
<dependency>
    <groupId>io.github.palexdev</groupId>
    <artifactId>mfxresources</artifactId>
    <version>11.1.0</version>
</dependency>
```

```groovy
implementation 'io.github.palexdev:mfxresources:11.1.0'
```

Edit: the artifact may still be unusable at the moment as its dependency MFXEffects is still to be published, will do soon

**Comment by stefanofornari** [comment]
Thanks for the update and the great work. Any chances the different modules will have maven build?

**Reply by palexdev** [reply]
You mean instead of Gradle?

**Reply by stefanofornari** [reply]
or in addition to... in most cases it would be just adding a pom.xml

**Reply by palexdev** [reply]
Hmm, sorry but that won't happen
I don't want having to tinker with Maven too

**Comment by stefanofornari** [comment]
What if I commit to maintain it?
Ste

**Comment by stefanofornari** [comment]
One thing about modularization... If you are planning to use java module system (which I would strongly discourage :) ), please take care of the cycling dependency VirtualizedFX <-> MaterialFX
